cclib/bridge/cclib2pyquante.py

```diff
@@ -9,23 +9,52 @@
 
 from cclib.parser.utils import find_package
 
+# Support for old PyQuante may be dropped in future release.
 _found_pyquante = find_package("PyQuante")
+_found_pyquante2 = find_package("pyquante2")
 if _found_pyquante:
     from PyQuante.Molecule import Molecule
+if _found_pyquante2:
+    from pyquante2 import molecule
 
 
-def _check_pyquante(found_pyquante):
-    if not found_pyquante:
+def _check_pyquante():
+    if not _found_pyquante and not _found_pyquante2:
         raise ImportError("You must install `PyQuante` to use this function")
 
 
-def makepyquante(atomcoords, atomnos, charge=0, mult=1):
-    """Create a PyQuante Molecule."""
-    _check_pyquante(_found_pyquante)
-    return Molecule("notitle",
-                    list(zip(atomnos, atomcoords)),
-                    units="Angstrom",
-                    charge=charge, multiplicity=mult)
+def makepyquante(data):
+    """Create a PyQuante Molecule from ccData object."""
+    _check_pyquante()
+
+    # Check required attributes.
+    required_attrs = {"atomcoords", "atomnos", "charge", "mult"}
+    missing = [x for x in required_attrs if not hasattr(data, x)]
+
+    if missing:
+        missing = " ".join(missing)
+        raise MissingAttributeError(
+            "Could not create pyquante molecule due to missing attribute: " + missing
+        )
+
+    # For older PyQuante
+    if _found_pyquante:
+        return Molecule(
+            "notitle",
+            [(data.atomnos[i], tuple(data.atomcoords[-1][i])) for i in range(len(data.atomnos))], 
+            units="Angstroms",
+            charge=data.charge,
+            multiplicity=data.mult,
+        )
+
+    # For pyquante2
+    elif _found_pyquante2:
+        return molecule(
+            [(data.atomnos[i], data.atomcoords[-1][i][0], data.atomcoords[-1][i][1], data.atomcoords[-1][i][2]) for i in range(len(data.atomnos))], 

```

Comment:
Build this up more incrementally in a few lines, so that it's more readable.

Reply:
I've split this into two lines. Thanks.